@plotly/dash-test-components/src/components/MyPersistedComponent.js

```diff
@@ -0,0 +1,293 @@
+import React, {PureComponent} from 'react';
+import PropTypes from 'prop-types';
+
+// simple function to substitute is-numeric for our use case
+const isNumber = (n) => {
+    return !isNaN(parseFloat(n)) && isFinite(n);
+}
+
+// eslint-disable-next-line no-implicit-coercion
+const convert = val => (isNumber(val) ? +val : NaN);
+
+const isEquivalent = (v1, v2) => v1 === v2 || (isNaN(v1) && isNaN(v2));
+
+// using these inline functions instead of ramda
+const isNil = val => val == null
+
+const omit = (key, obj) => {
+    const { [key]: omitted, ...rest } = obj;
+    return rest;
+  }
+
+/**
+ * Adapted dcc input component for persistence tests.
+ *
+ * Note that some unnecessary props have been removed.
+ */
+export default class MyPersistedComponent extends PureComponent {
+    constructor(props) {
+        super(props);
+
+        this.input = React.createRef();
+
+        this.onChange = this.onChange.bind(this);
+        this.onEvent = this.onEvent.bind(this);
+        this.onKeyPress = this.onKeyPress.bind(this);
+        this.setInputValue = this.setInputValue.bind(this);
+        this.setPropValue = this.setPropValue.bind(this);
+    }
+
+    UNSAFE_componentWillReceiveProps(nextProps) {
+        const {value} = this.input.current;
+        const valueAsNumber = convert(value);
+        this.setInputValue(
+            isNil(valueAsNumber) ? value : valueAsNumber,
+            nextProps.value
+        );
+        if (this.props.type !== 'number') {
+            this.setState({value: nextProps.value});
+        }
+    }
+
+    componentDidMount() {
+        const {value} = this.input.current;
+        const valueAsNumber = convert(value);
+        this.setInputValue(
+            isNil(valueAsNumber) ? value : valueAsNumber,
+            this.props.value
+        );
+    }
+
+    UNSAFE_componentWillMount() {
+        if (this.props.type !== 'number') {
+            this.setState({value: this.props.value});
+        }
+    }
+
+    render() {
+        const valprops =
+            this.props.type === 'number' ? {} : {value: this.state.value};
+        const {loading_state} = this.props;
+        return (
+            <input
+                data-dash-is-loading={
+                    (loading_state && loading_state.is_loading) || undefined
+                }
+                ref={this.input}
+                onChange={this.onChange}
+                onKeyPress={this.onKeyPress}
+                {...valprops}
+                {...omit(
+                    [
+                        'debounce',
+                        'value',
+                        'n_submit',
+                        'n_submit_timestamp',
+                        'selectionDirection',
+                        'selectionEnd',
+                        'selectionStart',
+                        'setProps',
+                        'loading_state',
+                    ],
```

Comment:
As we had discussed, this is copied over from `dcc.Input`. A lot of these props are not really useful for the test either. Try and keep the surface area for this component to the minimum you need to do the tests you need, nothing more.

Reply:
37d454f